src/queries/searchResultQuery.ts

```diff
@@ -0,0 +1,31 @@
+class SearchResultQuery {
+  queryParams: any;
+  searchResultQuery: any;
+
+  constructor(queryParams: any = {}) {
+    this.queryParams = queryParams;
+    this.searchResultQuery = {};
+  }
+
+  construct() {
```

Comment:
That is an odd name since it is very close to `constructor` 💣  How about simply:

```suggestion
  build() {
```